languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/GermanSpellerRule.java

```diff
@@ -2247,80 +2260,65 @@ protected boolean ignorePotentiallyMisspelledWord(String word) throws IOExceptio
-        // nonStrictSplitter case, it splits like "[Priorität, sdings]", we fix that here to match the strict splitter case:
-        part1 = part1 + "s";
-        part2 = part2.substring(1);
-        hasInfixS = true;
-      }
-    } else if (parts.size() == 3 && parts.get(1).equals("s") && word.contains("-") && startsWithUppercase(parts.get(2))) {
-      // e.g. "Prioritäts-Dings" gets split like "Priorität", "s", "dings" -> treat it as if there was no "-":
-      part1 = parts.get(0) + "s";
-      part2 = lowercaseFirstChar(parts.get(2));
-      hasInfixS = true;
-    } else if (parts.size() == 3 && !word.contains("-")) {
-      // e.g. Hundefutterschachtel = Hunde, Futter, Schachtel
-      part1 = parts.get(0);
-      part2 = parts.get(1);
-      String compound1 = parts.get(0) + parts.get(1);
-      String compound1noS = compound1.replaceFirst("s$", "");
-      String compound2 = uppercaseFirstChar(parts.get(1)) + parts.get(2);
-      boolean compound1ok = false;
-      if (germanPrefixes.contains(part2)) {
-        compound1ok = 
-          (((!isMisspelled(part1) && !isMisspelled(part1+parts.get(2))) ||  // Weinkühlschrank gets split into Wein, kühl, schrank
-          ignorePotentiallyMisspelledWord(part1+parts.get(2))) &&
-          parts.get(2).length() >= 3) ||  //Vorraus --> Vor, rau, s
-          (!isMisspelled(compound1) || ignorePotentiallyMisspelledWord(compound1) ||   //Menschenrechtsdemos as 'rechts' is in germanPrefixes
-          !isMisspelled(compound1noS) || ignorePotentiallyMisspelledWord(compound1noS)); 
-      } else {
-        compound1ok =
-          !isMisspelled(compound1) || ignorePotentiallyMisspelledWord(compound1) ||
-          !isMisspelled(compound1noS) || ignorePotentiallyMisspelledWord(compound1noS);
+
+      // Make sure that the individual parts of the compound have appropriate length.
+      // Short words can also be typos.
+      if (!isValidPartLength(part1, part2)) {
+        return false;
       }
-      boolean compound2ok =
-        (!isMisspelled(compound2) || ignorePotentiallyMisspelledWord(compound2)) && isNoun(compound2);
-      return compound1ok && compound2ok;
+
+      return processTwoPartCompounds(part1, part2);
+
+    } else if (parts.size() == 3) {
+      return processThreePartCompound(parts);
     } else {
       // more than three parts can be supported later
       return false;
     }
-    if (word.contains("-" + part2)) {
-      // don't accept e.g. "Implementierungs-pflicht"
-      return false;
-    }
+  }
+
+  private boolean isValidWordLength(String word) {
+    return word.length() <= MIN_WORD_LENGTH || word.length() >= MAX_WORD_LENGTH;
+  }
+
+  private boolean isValidPartLength(String part1, String part2) {
     // don't assume very short parts (like "Ei") are correct, these can easily be typos:
-    if ((hasInfixS || part1.endsWith("s")) && part1.length() >= 4 /* includes 's' */ && part2.length() >= 3 && startsWithLowercase(part2)) {
```

Comment:
Warum fällt das hier raus? Dadurch, dass nicht geprüft wird, ob part2 bereits mit uppercase reingegeben wird, funktioniert in der GermanSpellerRuleTest.java dieser case nicht mehr: 
`assertFalse(rule.ignorePotentiallyMisspelledWord("HaltungsSchäden"));`

Reply:
Dort habe ich letzte Woche aufgehört. Da es bisher keine systematische Lösung für Bindestriche gibt, habe ich solche Fälle vorerst weggelassen. Denn abhängig davon, ob es einen Bindestrich gibt, können Komposita korrekt sein ("Haltungs-Schäden" vs "HaltungsSchäden")

Reply:
Es können natürlich striktere Bedingungen an die Bindestrichsetzung gelegt werden, aber das besprechen wir ja noch

Reply:
CamelCase für zweiteilige Komposita failt wieder